docs/core/whats-new/dotnet-9/runtime.md

```diff
@@ -106,9 +118,82 @@ static int Sum(int[] arr)
 
 The index variable, `i`, is 4 bytes in size. At the assembly level, 64-bit registers are typically used to hold array indices on x64, and in previous .NET versions, the compiler generated code that zero-extended `i` to 8 bytes for the array access, but continued to treat `i` as a 4-byte integer elsewhere. However, extending `i` to 8 bytes requires an additional instruction on x64. With IV widening, the 64-bit JIT compiler now widens `i` to 8 bytes throughout the loop, omitting the zero extension. Looping over arrays is very common, and the benefits of this instruction removal quickly add up.
 
+#### Post-indexed addressing on Arm64
+
+Index variables are frequently used to read sequential regions of memory. Consider the idiomatic `for` loop:
+
+```csharp
+int sum = 0;
+int[] nums = [0..12];
+
+for (int i = 0; i < nums.Length; i++)
+{
+    sum += nums[i];
+}
+```
+
+For each iteration of the loop, the index variable `i` is used to read an integer in `nums`, and then `i` is incremented. In Arm64 assembly, these two operations look as follows:
+
+```al
+ldr w0, [x1]
+add x1, x1, #4
+```
+
+`ldr w0, [x1]` loads the integer at the memory address in `x1` into `w0`; this corresponds to the access of `nums[i]` in the source code. Then, `add x1, x1, #4` increases the address in `x1` by four bytes, moving to the next integer in `nums`. This instruction corresponds to the `i++` operation executed at the end of each iteration.
+
+Arm64 supports post-indexed addressing, where the "index" register is automatically incremented after its address is used. This means that two instructions can be combined into one, making the loop more efficient. The CPU only needs to decode one instruction instead of two, and the loop's code is now more cache-friendly.
+
+Here's what the updated assembly looks like:
+
+```al
+ldr w0, [x1], #0x04
+```
+
+The `#0x04` at the end means the address in `x1` is incremented by four bytes after it's used to load an integer into `w0`. The 64-bit compiler now uses post-indexed addressing when generating Arm64 code. While x64 doesn't support post-indexed addressing, [induction variable widening](#induction-variable-widening) is similarly useful for optimizing memory accesses with loop-index variables.
+
+#### Strength reduction
```

Comment:
It isn't immediately obvious how these two optimizations make sense together. They are optimizing the same (very small) sample differently. I read that the former is for Arm64 and the latter is for x64. That is called out but didn't pop for me at first. It also didn't pop that they are effectively achieving the same thing. The former is using an ISA capability and the latter runtime capabilities.

The mention of the x64-specific induction capability in the Arm64 section as being "similarly useful" reads as misleading to me since it is achieving something quite different (per my understanding) while strength reduction (the next feature in this doc) is more similar.

Is this a fair analysis @amanasifkhalid ?

Reply:
I think that's fair. Since we're describing both IV widening and post-indexed addressing on the same page, perhaps we can remove this call-out altogether? The explanations already mention which optimization is x64-specific, ARM64-specific, etc.

Reply:
That sounds good. I was the one who asked for the call-out in the first place, but now that I better understand the content, it no longer makes sense to me. Also, I'm using a slightly different "brain filter" here since our docs are intended to readable by a much larger audience.

I think the following should be made clear:

- The two optimizations (described by the two sections) are similar, however, we took different approaches since Arm64 offers a CPU capability and x64 doesn't.
- For x64, we additionally implemented the induction feature, again, due to differences in CPU/ISA capability/needs.